File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.ts
function minCostClimbingStairs(cost: number[]): number {
    let firstStep = cost[0], secondStep = cost[1];
    
    for(let i = 2; i< cost.length; i++) {
        const tmp = Math.min(firstStep, secondStep) + cost[i];
        firstStep = secondStep;
        secondStep = tmp;
    }
    return Math.min(firstStep, secondStep) ;
};

File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        dp = [float('inf')]*len(cost)
        
        for i in range(len(cost) - 1, -1, -1):
          if i + 2 >= len(cost):
            dp[i] = cost[i]
          else:
            dp[i] = cost[i] + min(dp[i+1], dp[i+2])
        
        return min(dp[0], dp[1])